### PR TITLE
font-adobe-source-code-pro: update to 2.038R.

### PR DESCRIPTION
**Pull request by travankor**
#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [X] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

Should the transitional dummy package be removed?

### CODE REVIEW

File: srcpkgs/font-adobe-source-code-pro/template [hunk 1]
```diff
@@ -1,21 +1,25 @@
 # Template file for 'font-adobe-source-code-pro'
 pkgname=font-adobe-source-code-pro
-version=2.030R
-revision=3
-wrksrc="source-code-pro-${version}-ro-1.050R-it"
+version=2.038R
+revision=1
+wrksrc="source-code-pro-${version}-ro-1.058R-it-1.018R-VAR"
```
Comment (sgn) [inline srcpkgs/font-adobe-source-code-pro/template:5]: Change like this make me think it's better to just use `create_wrksrc=yes` then make its own version of `do_extract` to `bsdtar -x --strip-components=1`

File: srcpkgs/font-adobe-source-code-pro/template [hunk 2]
```diff
@@ -1,21 +1,25 @@
...
 font_dirs="/usr/share/fonts/OTF/"
 
 do_install() {
 	for i in ${wrksrc}/OTF/*.otf; do
 		vinstall $i 644 usr/share/fonts/OTF/
 	done
+
+	for i in ${wrksrc}/VAR/*.otf; do
+		vinstall $i 644 usr/share/fonts/OTF/
+	done
```
Comment (sgn) [inline srcpkgs/font-adobe-source-code-pro/template:22]: Perhap:
```sh
for i in ${wrksrc}/*/*.otf; do
```

**Comment by ericonr** [comment]
@sgn satisfied or still need to add the custom `do_extract`?

**Comment by sgn** [comment]
I don't care that much.
I look into the template once again, I think it's better to have 2 variables for italic and var version.
I don't know :shrug:

**Comment by sgn** [comment]
I'm thinking about something like this:

```diff
diff --git a/srcpkgs/font-adobe-source-code-pro/template b/srcpkgs/font-adobe-source-code-pro/template
index b40277d7aa..a58d81e5fd 100644
--- a/srcpkgs/font-adobe-source-code-pro/template
+++ b/srcpkgs/font-adobe-source-code-pro/template
@@ -1,14 +1,15 @@
 # Template file for 'font-adobe-source-code-pro'
 pkgname=font-adobe-source-code-pro
-version=2.038R
+version=2.038R~ro+1.058R~it+1.018R~VAR
 revision=1
-wrksrc="source-code-pro-${version}-ro-1.058R-it-1.018R-VAR"
+_version="${version//\~/-}"
+wrksrc="source-code-pro-${_version//+/-}"
 depends="font-util"
 short_desc="Monospaced font family for user interface and coding environments"
 maintainer="Piero La Terza <platerza@protonmail.com>"
 license="OFL-1.1"
 homepage="https://adobe-fonts.github.io/source-code-pro/"
-distfiles="https://github.com/adobe-fonts/source-code-pro/archive/${version}-ro/1.058R-it/1.018R-VAR.tar.gz"
+distfiles="https://github.com/adobe-fonts/source-code-pro/archive/${_version//+//}.tar.gz"
 checksum=d8f879fefd66ecb7291ea2d73984b9c0a97fb499424e58290ba6b81787d0c725
 font_dirs="/usr/share/fonts/OTF/"
 
```

**Comment by sgn** [comment]
Or even this:
```diff
diff --git a/srcpkgs/font-adobe-source-code-pro/template b/srcpkgs/font-adobe-source-code-pro/template
index b40277d7aa..b804f0c916 100644
--- a/srcpkgs/font-adobe-source-code-pro/template
+++ b/srcpkgs/font-adobe-source-code-pro/template
@@ -1,19 +1,21 @@
 # Template file for 'font-adobe-source-code-pro'
 pkgname=font-adobe-source-code-pro
-version=2.038R
+version=2.038R~ro+1.058R~it
 revision=1
-wrksrc="source-code-pro-${version}-ro-1.058R-it-1.018R-VAR"
+_version="${version//\~/-}"
+create_wrksrc=yes
 depends="font-util"
 short_desc="Monospaced font family for user interface and coding environments"
 maintainer="Piero La Terza <platerza@protonmail.com>"
 license="OFL-1.1"
 homepage="https://adobe-fonts.github.io/source-code-pro/"
-distfiles="https://github.com/adobe-fonts/source-code-pro/archive/${version}-ro/1.058R-it/1.018R-VAR.tar.gz"
-checksum=d8f879fefd66ecb7291ea2d73984b9c0a97fb499424e58290ba6b81787d0c725
+distfiles="https://github.com/adobe-fonts/source-code-pro/archive/${_version//+//}.tar.gz"
+distfiles="https://github.com/adobe-fonts/source-code-pro/releases/download/${_version//+/%2F}%2F1.018R-VAR/OTF-source-code-pro-${_version//+/-}.zip"
+checksum=478028ec70adc1ff73848a546c3ad266716a0096cd1f1a1ef18aeff0199d5996
 font_dirs="/usr/share/fonts/OTF/"
 
 do_install() {
-	for i in ${wrksrc}/*/*.otf; do
+	for i in SourceCodePro-*.otf; do
 		vinstall $i 644 usr/share/fonts/OTF/
 	done
 }
```

**Pull request merged**